src/model_repository_manager.cc

```diff
@@ -1703,14 +1700,28 @@ ModelRepositoryManager::RepositoryIndex(
   std::set<std::string> seen_models;
   std::set<std::string> duplicate_models;
   for (const auto& repository_path : repository_paths_) {
+    // For any mapped models in this repository, save the mapping
+    // from their subdirectory name to model name.
+    std::map<std::string, std::string> models_in_repo;
+    for (const auto& mapping_it : model_mappings_) {
+      if (mapping_it.second.first == repository_path) {
+        models_in_repo.emplace(mapping_it.second.first, repository_path);
```

Comment:
I think this should be `models_in_repo.emplace(BaseName(mapping_it.second.second), mapping_it.first);` (`key->directory_name`; `value->override_name`). In line 1715, you want to search if the directory name is a key of `models_in_repo`, and set `model` to the corresponding value.

Reply:
Thanks for catching this. So many changes to the mapping design haha.

As per your comment above, I made a follow-up ticket to test the model indexes for mapped models.